packages/@o3r/core/schematics/component/container/index.ts

```diff
@@ -70,45 +72,226 @@ export function ngGenerateComponentContainer(options: NgGenerateComponentContain
+        path: componentDestination,
+        name: properties.componentName,
+        inlineStyle: false,
+        inlineTemplate: false,
+        viewEncapsulation: 'None',
+        changeDetection: 'OnPush',
+        style: 'none',
+        type: 'Component',
+        skipSelector: false,
+        skipTests: false,
+        standalone: options.standalone,
+        skipImport: true,
+        flat: true
       }),
-      renameTemplateFiles(),
-      move(componentDestination)
-    ]), MergeStrategy.Overwrite));
+      // Angular schematics generate spec file with this pattern: component-name.component.spec.ts
+      move(ngSpecPath, o3rSpecPath),
+      // Angular schematics generate template file with this pattern: component-name.component.html
+      chain([
+        move(ngTemplatePath, o3rTemplatePath),
+        (t) => {
+          t.overwrite(
+            componentPath,
+            t.readText(componentPath).replace(
+              path.basename(ngTemplatePath),
+              path.basename(o3rTemplatePath)
+            )
+          );
+          return t;
+        }
+      ]),
+      schematic('convert-component', {
+        path: componentPath,
+        skipLinter: options.skipLinter
+      })
+    );
 
     if (!options.standalone) {
-      rules.push(mergeWith(apply(url(MODULE_TEMPLATE_PATH), [
-        template({
-          ...properties
+      rules.push(
+        externalSchematic('@schematics/angular', 'module', {
+          project: properties.projectName,
+          path: componentDestination,
+          flat: true,
+          name: properties.componentName
         }),
-        renameTemplateFiles(),
-        move(componentDestination)
-      ]), MergeStrategy.Overwrite));
+        (t) => {
+          const modulePath = path.join(componentDestination, `${properties.name}.module.ts`);
+          const moduleSourceFile = ts.createSourceFile(
+            modulePath,
+            t.readText(modulePath),
+            ts.ScriptTarget.ES2020,
+            true
+          );
+          const changes = addDeclarationToModule(moduleSourceFile, modulePath, `${properties.componentName}Component`, `./${properties.name}.component`);
+          const recorder = t.beginUpdate(modulePath);
+          applyToUpdateRecorder(recorder, changes);
+          t.commitUpdate(recorder);
+          return t;
+        },
+        (t) => {
+          const modulePath = path.join(componentDestination, `${properties.name}.module.ts`);
+          const moduleSourceFile = ts.createSourceFile(
+            modulePath,
+            t.readText(modulePath),
+            ts.ScriptTarget.ES2020,
+            true
+          );
+          const changes = addExportToModule(moduleSourceFile, modulePath, `${properties.componentName}Component`, `./${properties.name}.component`);
+          const recorder = t.beginUpdate(modulePath);
+          applyToUpdateRecorder(recorder, changes);
+          t.commitUpdate(recorder);
+          return t;
```

Comment:
couldn't you use the `--module` option of the component generator to do that ?

Reply:
the `--module` is to expose the component in an existing module 
it's not working if the module doesn't exist